### ISSUE TITLE
Hat ID EEPROM

### ISSUE THREAD

**Issue by mariusmotea** [m1geo/Pi5_PCIe#4]
Hi,

I notice you connect the i2c pins to SMDAT and SMCLK but i think most likely these must be connected to an i2c eeprom chip.

https://hackaday.io/project/167888-not-quite-useless-raspberry-pi-replacement/log/170018-raspberry-pi-hat-id-eeprom-details

https://github.com/raspberrypi/hats

```
A board can only be called a HAT if:

It conforms to the basic add-on board requirements
It has a valid ID EEPROM (including vendor info, GPIO map and valid device tree information).
```

**Comment by m1geo** [comment]
The HAT ID EEPROM (including vendor info, GPIO map and valid device tree information) is connected to [GPIO1 pin28](https://pinout.xyz/pinout/pin28_gpio1/) and [GPIO0 pin27](https://pinout.xyz/pinout/pin27_gpio0/). **I don't implement this.**

The PCIe_SMDAT and PCIe_SMCLK come via the 16pin FPC cable. they are different I2C controllers. They're likely used for [IPMI based activities](https://en.wikipedia.org/wiki/Intelligent_Platform_Management_Interface) and [SMBus](https://en.wikipedia.org/wiki/System_Management_Bus) and side-channel data.

![image](https://github.com/m1geo/Pi5_PCIe/assets/758615/cb1a2a14-70db-4fc2-a4b8-766911c47bf1)

**TLDR: There are many different I2C controllers on the Pi. We're talking about different ones. Hope that helps.**

**Comment by mikegapinski** [comment]
@m1geo Did you find a working dtoverlay that enables i2c on the 16 pin connector? They removed the reference to it on the official doc but I'd be very happy if it is still there. 

We have a test board where we connected the EEPROM used for HAT ID to the fpc connector. In theory we could "expand" the upcoming HAT+ standard to boards that are not using GPIO at all.

**Comment by m1geo** [comment]
No, I didn't, but the 16p header isn't for the HAT EEPROM. The official spec (see main readme) doesn't list the I2C, so I guess there's no offical support. The HAT EEPROM will need to connect to the GPIO header EEPROM pins.

**Comment by mariusmotea** [comment]
Pineberry has released two pcie adapters one using the 40 pin header and one using only the PCIe 16 pin connector. I looked at the pics on the one that use only the PCIe connector and the i2c pins go to two components, one looks to be the eeprom chip and the second is an i2c current measurement.

From specs:

 - Dedicated I2C EEPROM Memory: Fully compatible with HAT ID for advanced functionality and configuration.
 - Advanced Power Monitoring and Diagnostics: Continuous, real-time monitoring and diagnostic system for voltage, current, and power parameters via an I2C interface.

**Comment by mikegapinski** [comment]
I can't say too much about this yet but we are planning on using the I2C on the HatDrive Bottom that does not use the GPIO directly. The current monitoring feature is very important to us and we found a decently elegant solution that will make it work without any hardware changes on the boards that are currently shipping out. 

We do however ship both the boards in the first batch with an empty EEPROM since the payload for auto detection is not yet public, we'll release an update script that'll make it work as soon as we know the details. Right now you need to use the PCIE_PROBE bootloader config flag to enable booting from NVMe. That is what the HAT+ spec will bring. 

I don't understand why:
1. PCIe enumeration is not happening by default if there is no SD present. What is the point of changing the config to enable this?
2. I2C on the 16pin FPC evaporated from the final spec, it is connected/present in the SOC but not supported in the firmware. HABs or PIPs (or whatever you decide to call them) are better than standard NVMe HATs that go over the GPIO in so many ways. We also see more demand for them, and the upcoming competition (Pimoroni etc) also went with this approach. I think that they'll add ID detection for those type of designs at some point, it just makes sense.

If "PCIe_SMDAT" and "PCIe_SMCLK" are indeed GPIO pins that are coming from the 2712 directly there should be a way to make them appear as a I2C device with a overlay / device tree config change. I'd love to have the time to play around with the kernel more but we are swamped with orders. I'll get this figured out one day I hope.

**Comment by m1geo** [comment]
See #1 for I2C pinout on the FPC 16-pin connector.
